parla/task_runtime.py

```diff
@@ -847,24 +807,39 @@ def enqueue_task(self, task):
         #     self._push_task(task)
         # This would need to fail over to the scheduler level enqueue if the resources is not available for assignment.
 
-    def _enqueue_task_local(self, task):
+    def assign_task(self, task):
         with self._monitor:
-            self._queue.appendleft(task)
+            if self.task:
+                raise WorkerThreadException("Tried to assign task to WorkerThread that already had one.")
+            self.task = task
             self._monitor.notify()
 
+    def _remove_task(self):
+        with self._monitor:
+            if not self.task:
+                raise WorkerThreadException("Tried to remove a nonexistent task.")
+            self.task = None
+
     def run(self) -> None:
         try:
             with self:
                 for component in self.scheduler.components:
                     component.initialize_thread()
                 while self._should_run:
                     self._status = "Getting Task"
-                    task: Task = self._pop_task()
-                    if not task:
-                        break
-                    print("Worker thread start:", task.name, sep='')
-                    self._status = "Running Task {}".format(task)
-                    task.run()
+                    if not self.task:
+                        logger.debug("Blocking for a task: %r (%s)", self, self._monitor)
+                        print("[WorkerThread ", self.index,"] Blocking for a task.", sep='')
+                        with self._monitor:
+                            self._monitor.wait()
+                        print("[WorkerThread ", self.index,"] Waking up.", sep='')
+                    if not self.task:
+                        break # TODO: Properly raise exception here
+                    print(f"[WorkerThread {self.index}] Starting:", self.task.name, sep='')
+                    self._status = "Running Task {}".format(self.task)
+                    self.task.run()
+                    self._remove_task()
+                    self.scheduler.free_thread(self)
```

Comment:
This code is basically copied from here https://github.com/ut-parla/Parla.py/pull/99/commits/97350f80965e4c3a14f3a59e36d2c438f6a0ff44#diff-69c8f130075278971cac4616ec977714100e803d919c4a099059a572fb28201dL796-L812

but it supports a single task instead of a queue